flink-end-to-end-tests/test-scripts/common_mesos_docker.sh

```diff
@@ -72,7 +73,25 @@ function build_image() {
 function wait_job_terminal_state_mesos {
   local job=$1
   local expected_terminal_state=$2
-  wait_job_terminal_state $1 $2 "mesos-appmaster"
+  local successful_response_regex="\{.*\"application-status\":\"${expected_terminal_state}\".*\}"
+  local query_url="${REST_PROTOCOL}://${NODENAME}:8081/jobs/${job}/execution-result"
+  local TIMEOUT=120
+
+  echo "Waiting for job ($job) to reach terminal state $expected_terminal_state ..."
+
+  for i in $(seq 1 ${TIMEOUT}); do
+    QUERY_RESULT=$(curl "$query_url" 2> /dev/null || true)
+    # ensure the response adapts with the suceessful regex
```

Comment:
Not sure how useful this comment is. Also, typo: *successful*

If this isn't needed for FLINK-15417, I would prefer to create a separate issue and apply this to standalone mode as well. WDYT?

Reply:
Yes, it is. I create FLINK-15870 to track this. FYI.